scanner/scanner.go

```diff
@@ -432,8 +445,41 @@ func scanGlobalVar(s *Scanner) (token.Token, []byte) {
 	return token.IdentGlobalVar, s.src[s.begin:s.offset]
 }
 
-func scanMod(s *Scanner) (token.Token, []byte) {
-	if s.char == '=' { // %=
+func closeBracket(c byte) byte {
+	switch c {
+	case '{':
+		return '}'
+	case '(':
+		return ')'
+	case '[':
+		return ']'
+	case '<':
+		return '>'
+	}
+	return 0
+}
+
+func scanPercent(s *Scanner) (token.Token, []byte) {
```

Comment:
__[gometalinter]__
* [gocyclo] cyclomatic complexity 13 of function scanPercent() is high (> 10)

<!-- Comment by [SideCI](https://sideci.com) -->